src/spikeinterface/sortingcomponents/benchmark/benchmark_matching.py

```diff
@@ -16,101 +17,483 @@
+
+        Parameters
+        ----------
+        we: WaveformExtractor
+            The new WaveformExtractor.
+        template_mode: {'mean' | 'median' | 'std'}
+            The mode to use to extract templates from the WaveformExtractor. (Default: 'median')
+
+        Returns
+        -------
+        methods_kwargs: dict
+            A dictionary of method_kwargs for each method.
+        """
+        templates = we.get_all_templates(we.unit_ids, mode=template_mode)
+        methods_kwargs = copy.deepcopy(self.methods_kwargs)
+        for method in self.methods:
+            method_kwargs = methods_kwargs[method]
+            if method == "wobble":
+                method_kwargs.update(dict(templates=templates, nbefore=we.nbefore, nafter=we.nafter))
+            else:
+                method_kwargs["waveform_extractor"] = we
+        return methods_kwargs
 
-    def run(self):
-        t_start = time.time()
-        self.spikes = find_spikes_from_templates(
-            self.recording, method=self.method, method_kwargs=self.method_kwargs, **self.job_kwargs
+    def run_matching_misclassed(
+        self, fraction_misclassed, min_similarity=-1, seed=0, we_kwargs=None, template_mode="median"
+    ):
+        """Run template matching with a given fraction of misclassified spikes.
+
+        Parameters
+        ----------
+        fraction_misclassed: float
+            The fraction of misclassified spikes.
+        min_similarity: float
+            The minimum cosine similarity between templates to be considered similar. (Default: -1)
+        seed: int
+            Random seed. (Default: 0)
+        we_kwargs: dict
+            A dictionary of keyword arguments for the WaveformExtractor.
+        template_mode: {'mean' | 'median' | 'std'}
+            The mode to use to extract templates from the WaveformExtractor. (Default: 'median')
+
+        Returns
+        -------
+        sortings: dict
+            A dictionary that maps method --> NumpySorting.
+        gt_sorting: NumpySorting
+            The ground-truth sorting used for template matching (with misclassified spike trains).
+        """
+        try:
+            assert 0 <= fraction_misclassed <= 1
+        except AssertionError:
+            raise ValueError("'fraction_misclassed' must be between 0 and 1.")
+        try:
+            assert -1 <= min_similarity <= 1
+        except AssertionError:
+            raise ValueError("'min_similarity' must be between -1 and 1.")
+        if we_kwargs is None:
+            we_kwargs = {}
+        we_kwargs.update(dict(seed=seed, overwrite=True, load_if_exists=False, **self.job_kwargs))
+        rng = np.random.default_rng(seed)
+
+        # Randomly misclass spike trains
+        spike_time_indices, labels = [], []
+        for unit_index, unit_id in enumerate(self.we.unit_ids):
+            unit_sorting = self.gt_sorting.get_unit_spike_train(unit_id=unit_id)
```

Comment:
unit_sorting in the context of spikeinterface is a bit confusin.
spike_train is more easy to read no ?